activerecord/lib/active_record/connection_adapters/abstract/connection_pool/reaper.rb

```diff
@@ -7,12 +7,29 @@ module ConnectionAdapters
     class ConnectionPool
       # = Active Record Connection Pool \Reaper
       #
-      # Every +frequency+ seconds, the reaper will call +reap+ and +flush+ on
-      # +pool+. A reaper instantiated with a zero frequency will never reap
-      # the connection pool.
+      # The reaper is a singleton that exists in the background of the process
+      # and is responsible for general maintenance of all the connection pools.
```

Comment:
> is responsible for general maintenance of all the connection pools.

There's only one pool right?

If so, could we say "maintenance of the connections"? This is subtle, but I think using a singular pool term here helps because we do also run multiple instances of an application where each has its own pool, and within a pool we expect to have multiple connections.

Reply:
Maybe it's relevant with `multi_thread_pool` with [async queries](https://guides.rubyonrails.org/configuring.html#config-active-record-async-query-executor) ?